plugins/csp-wms-overlays/src/WebMapTextureLoader.hpp

```diff
@@ -0,0 +1,92 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_TEXTURE_LOADER_HPP
+#define CSP_WMS_TEXTURE_LOADER_HPP
+
+#include "WebMapLayer.hpp"
+#include "WebMapService.hpp"
+
+#include "../../../src/cs-utils/ThreadPool.hpp"
+
+#include <boost/filesystem.hpp>
+
+#include <array>
+#include <map>
+
+namespace csp::wmsoverlays {
+
+/// Struct for storing texture data along with some metadata.
+struct WebMapTexture {
+  unsigned char* mData;
+  int            mWidth;
+  int            mHeight;
+};
+
+/// Class for requesting map textures from Web Map Services.
+class WebMapTextureLoader {
+ public:
+  /// Struct for defining parameters for a request to a WMS.
+  struct Request {
+    int                        mMaxSize;
+    std::string                mStyle;
+    Bounds                     mBounds;
+    std::optional<std::string> mTime;
+  };
+
+  /// Creates a new ThreadPool with the specified amount of threads.
+  WebMapTextureLoader();
+
+  ~WebMapTextureLoader();
+
+  /// Async WMS texture loader.
+  /// Returns an empty optional if loading the texture failed.
+  std::future<std::optional<WebMapTexture>> loadTextureAsync(WebMapService const& wms,
+      WebMapLayer const& layer, Request const& request, std::string const& mapCache,
+      bool const& saveToCache);
```

Comment:
No need to pass built-ins as reference.

```suggestion
      bool saveToCache);
```